walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,109 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    boolean[] unavailableMinutes = new boolean[24 * 60];
+    Collection<String> meetingAttendees = request.getAttendees();
+    long meetingDuration = request.getDuration();
+
+    for (Event event : events) {
+      Set<String> eventAttendees = event.getAttendees();
+      for (String meetingAttendee : meetingAttendees) {
+        if (eventAttendees.contains(meetingAttendee)) {
+          TimeRange eventTimeRange = event.getWhen();
+          addEventTimeRange(unavailableMinutes, eventTimeRange);
+          break;
+        }
+      }
+    }
+    
+    Collection<TimeRange> availableTimeSlots =
+        checkForAvailableTimeSlots(unavailableMinutes, meetingDuration);
+
+    /*System.out.println("---> Unavailable Minutes <---");
+    for (int i = 0; i < 24; i++) {
+      System.out.print(i + ": ");
+      for (int j = 0; j < 60; j++) {
+        if (unavailableMinutes[(i * 60) + j]) System.out.print("1");
+        else System.out.print("0");
+      }
+      System.out.println();
+    }
+    System.out.println("---> Time Ranges <---");
+    for (TimeRange timeRange : availableTimeSlots) {
+      System.out.println("Start: " + timeRange.start() + " - End: " + timeRange.end());
+    }
+    System.out.println();*/
+
+    return availableTimeSlots;
+  }
+
+  /**
+   * Adds the time range of a specific event to the array of unavailable minutes.
+   * @param unavailableMinutes the minutes where other events are happening.
+   * @param eventTimeRange the time range of a specific event to be added to unavailableMinutes.
+   */
+  private void addEventTimeRange(boolean[] unavailableTimeSlots, TimeRange eventTimeRange) {
```

Comment:
Perhaps if this is only tracking unavailable time slots, that should be reflected in the method's name?

Reply:
Done!